src/pointer-analysis/value_set.cpp

```diff
@@ -69,10 +67,22 @@ value_sett::entryt &value_sett::get_entry(const entryt &e, const typet &type)
   else
     index=e.identifier;
 
-  std::pair<valuest::iterator, bool> r =
-    values.insert(std::pair<irep_idt, entryt>(index, e));
-
-  return r.first->second;
+  auto existing_entry = values.find(index);
+  if(existing_entry.has_value())
+  {
+    entryt replacement = *existing_entry;
+    if(add_to_sets)
+      make_union(replacement.object_map, new_values);
```

Comment:
We could check the return value of `make_union()` and only do the replacement when the object map has changed.

Reply:
Done.